src/deploy/s3-upload.js

```diff
@@ -105,6 +105,13 @@ module.exports = {
             params.Expires = 0;
         }
 
-        return util.promisify(s3Client.putObject.bind(s3Client))(params);
+        const callback = (err, data) => {
+            if (err) {
+                console.error(`s3 upload of ${bucket}/${key} failed, error:`, err);
```

Comment:
actually, please change to:
```
console.error(`s3 upload of ${bucket}/${key} failed, error: ${err}`);
```
i.e. the stack trace is not helpful here, only clutters the log

Reply:
I am thinking as well to rethrow the err, since only the first failure probably matters, the rest will be just repeats